src/plugins/arithmetics/add_operation.cpp

```diff
@@ -342,32 +389,302 @@ namespace phylanx { namespace execution_tree { namespace primitives
+                codename_));
+    }
+
+    primitive_argument_type add_operation::add2d2d_stretch_y(
+        arg_type&& lhs, arg_type&& rhs, int stretch_y) const
+    {
+        // Stretch lhs y axis
+        if (stretch_y == 1)
+        {
+            // Avoid overwriting references, avoid memory
+            // reallocation when possible
+            if (rhs.is_ref())
+            {
+                blaze::DynamicMatrix<double> result(
+                    rhs.dimension(0), rhs.dimension(1));
+
+                for (std::size_t i = 0; i != result.columns(); ++i)
+                {
+                    blaze::column(result, i) =
+                        blaze::column(lhs.matrix(), 0) +
+                        blaze::column(rhs.matrix(), i);
+                }
+                return primitive_argument_type{std::move(result)};
+            }
+            else
+            {
+                for (std::size_t i = 0; i != rhs.dimension(1); ++i)
+                {
+                    blaze::column(rhs.matrix(), i) =
+                        blaze::column(lhs.matrix(), 0) +
+                        blaze::column(rhs.matrix(), i);
+                }
+                return primitive_argument_type{std::move(rhs)};
+            }
         }
+        // Stretch rhs y axis
+        else if (stretch_y == 2)
+        {
+            // Avoid overwriting references, avoid memory
+            // reallocation when possible
+            if (lhs.is_ref())
+            {
+                blaze::DynamicMatrix<double> result(
+                    lhs.dimension(0), lhs.dimension(1));
 
-        return primitive_argument_type{std::move(lhs)};
+                for (std::size_t i = 0; i != result.columns(); ++i)
+                {
+                    blaze::column(result, i) =
+                        blaze::column(lhs.matrix(), i) +
+                        blaze::column(rhs.matrix(), 0);
+                }
+                return primitive_argument_type{std::move(result)};
+            }
+            else
+            {
+                for (std::size_t i = 0; i != lhs.dimension(1); ++i)
+                {
+                    blaze::column(lhs.matrix(), i) =
+                        blaze::column(lhs.matrix(), i) +
+                        blaze::column(rhs.matrix(), 0);
+                }
+                return primitive_argument_type{std::move(lhs)};
+            }
+        }
+        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+            "add_operation::add2d2d_stretch_y",
+            execution_tree::generate_error_message(
+                "the dimensions of the operands do not match", name_,
+                codename_));
+    }
+
+    primitive_argument_type add_operation::add2d2d_stretch_xy(
+        arg_type&& lhs, arg_type&& rhs, int stretch_x, int stretch_y) const
+    {
+        blaze::DynamicMatrix<double> result(
+            (std::max)({lhs.dimension(0), rhs.dimension(0)}),
+            (std::max)({lhs.dimension(1), rhs.dimension(1)}));
```

Comment:
Why do you call the `std::max` algorithm (usable for arbitrary number of arguments) instead of the simple `std::max` function (for 2 arguments)?